File: app/src/test/java/me/rakshakhegde/dagger_android_injector/screens/main_screen/MainViewModelTest.java
package me.rakshakhegde.dagger_android_injector.screens.main_screen;

import org.junit.Test;

import me.rakshakhegde.dagger_android_injector.R;

import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.verify;

/**
 * Created by rakshakhegde on 27/04/17.
 */
public class MainViewModelTest {

	@Test
	public void showBtnClicked() throws Exception {
		// Setup
		MainView mockView = mock(MainView.class);

		// Wiring
		MainViewModel VM = new MainViewModel(mockView);
		VM.showBtnClicked();

		// Verify
		verify(mockView).showMessage(R.string.showing_toast);
	}

}

File: app/src/test/java/me/rakshakhegde/dagger_android_injector/screens/main_screen/MainViewModelTest.kt
package me.rakshakhegde.dagger_android_injector.screens.main_screen

import me.rakshakhegde.dagger_android_injector.R
import org.junit.Test
import org.mockito.Mockito.mock
import org.mockito.Mockito.verify

/**
 * Created by rakshakhegde on 27/04/17.
 */
class MainViewModelTest {

	@Test
	fun showBtnClicked() {
		// Setup
		val mockView = mock(MainView::class.java)

		// Wiring
		val VM = MainViewModel(mockView)
		VM.showBtnClicked()

		// Verify
		verify(mockView).showMessage(R.string.showing_toast)
	}

}
